src/Modal.php

```diff
@@ -29,176 +31,251 @@
+     *
+     * @param string $value The modal class.
+     *
+     * @return self
+     */
+    public function modalClass(string $value): self
+    {
         $new = clone $this;
-        $new->toggleButtonColor = $value;
+        $new->modalClass = $value;
 
         return $new;
     }
 
     /**
-     * Returns a new instance with the disabled toggle button.
+     * Returns a new instance with the specified modal content class.
+     *
+     * @param string $value The modal content class.
      *
      * @return self
      */
-    public function withoutToggleButton(): self
+    public function modalContentClass(string $value): self
     {
         $new = clone $this;
-        $new->withoutToggleButton = false;
+        $new->modalContentClass = $value;
 
         return $new;
     }
 
     /**
-     * Returns a new instance with the specified close button size.
+     * Returns a new instance with the specified toggle button options.
      *
-     * @param string $value The close button size.
+     * @param array $values Attribute values indexed by attribute names.
+     *
+     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
      *
      * @return self
      */
-    public function closeButtonSize(string $value): self
+    public function toggleButtonAttributes(array $values): self
     {
-        if (!in_array($value, self::SIZE_ALL, true)) {
-            $values = implode('"', self::SIZE_ALL);
-            throw new InvalidArgumentException("Invalid size. Valid values are: \"$values\".");
+        $new = clone $this;
+        $new->toggleButtonAttributes = $values;
+
+        return $new;
+    }
+
+    /**
+     * Returns a new instance with the specified toggle button color.
+     *
+     * @param string $value The toggle button color.
+     *
+     * @return self
+     */
+    public function toggleButtonColor(string $value): self
+    {
+        if (!in_array($value, self::COLORS, true)) {
+            $values = implode(' ', self::COLORS);
+            throw new InvalidArgumentException("Invalid color. Valid values are: $values.");
         }
 
         $new = clone $this;
-        $new->closeButtonSize = $value;
+        $new->toggleButtonColor = $value;
 
         return $new;
     }
 
     /**
-     * Returns a new instance with the specified close button options.
+     * Returns a new instance with the specified toggle button label.
      *
-     * @param array $value The close button options.
+     * @param string $value The toggle button label.
      *
-     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
+     * @return self
+     */
+    public function toggleButtonLabel(string $value): self
+    {
+        $new = clone $this;
+        $new->toggleButtonLabel = $value;
+
+        return $new;
+    }
+
+    /**
+     * Returns a new instance with the specified toggle button size.
+     *
+     * @param string $value The toggle button size.
```

Comment:
Accepted values should be listed here. Also, maybe we can remove `is-` prefix?

Reply:
done.